src/lib/Simplify.hs

```diff
@@ -103,6 +104,7 @@ simplifyAtom atom = case atom of
   Variant types label i value -> Variant <$>
     substEmbedR types <*> pure label <*> pure i <*> simplifyAtom value
   VariantTy items -> VariantTy <$> substEmbedR items
+  LabeledRow items -> LabeledRow <$> substEmbedR items
```

Comment:
I really wish we could beta-reduce those things directly in inference, so we don't have to think about them after that 😕

Reply:
Actually, we can probably delete this. I realized that `LabeledRow` atoms actually shouldn't exist past inference anyway.

Reply:
Oops, I'm wrong! If we have a previously-defined function with an implicit argument of type `LabeledRowKind`, the application of a `LabeledRow` atom to that function is preserved in the IR after inference (since it doesn't typecheck the already-defined function a second time). So we might see `LabeledRow` atoms until the end of `Simp`, then they go away once we've inlined the call.